### PR TITLE
Fixes Mind Blown / Magic Guard interaction and renames Steel Beam effect

### LINKED ISSUE #4033: Mind Blown + Magic Guard Interaction Bug
### Description

Pokemon with Magic Guard that have used Mind Blown or Steel Beam still take recoil damage when it is expected that Magic Guard stops it.

### Version

1.7.2 (Latest release)

### Upcoming/master Version

58102e6

### Discord contact info

._.terra._.

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Fixes  #4033

Mind Blown was the only one that needed fixing. Also renamed Steel Beam effect. Open to better name suggestions.

### CODE REVIEW

File: data/battle_scripts_1.s [hunk 1]
```diff
@@ -406,7 +406,7 @@ gBattleScriptsForMoveEffects::
 	.4byte BattleScript_EffectHit                     @ EFFECT_RISING_VOLTAGE
 	.4byte BattleScript_EffectHit                     @ EFFECT_BEAK_BLAST
 	.4byte BattleScript_EffectCourtChange             @ EFFECT_COURT_CHANGE
-	.4byte BattleScript_EffectSteelBeam               @ EFFECT_STEEL_BEAM
+	.4byte BattleScript_EffectMaxHp50Recoil               @ EFFECT_MAX_HP_50_RECOIL
```
Comment (Bassoonian) [inline data/battle_scripts_1.s:409]: indentation

File: include/constants/battle_move_effects.h [hunk 2]
```diff
@@ -383,7 +383,7 @@
 #define EFFECT_RISING_VOLTAGE               377
 #define EFFECT_BEAK_BLAST                   378
 #define EFFECT_COURT_CHANGE                 379
-#define EFFECT_STEEL_BEAM                   380
+#define EFFECT_MAX_HP_50_RECOIL                   380
```
Comment (Bassoonian) [inline include/constants/battle_move_effects.h:386]: indentation

File: src/battle_tv.c [hunk 3]
```diff
@@ -467,7 +467,7 @@ static const u16 sPoints_MoveEffect[NUM_BATTLE_MOVE_EFFECTS] =
     [EFFECT_RISING_VOLTAGE]             = 0, // TODO: Assign points
     [EFFECT_BEAK_BLAST]                 = 0, // TODO: Assign points
     [EFFECT_COURT_CHANGE]               = 0, // TODO: Assign points
-    [EFFECT_STEEL_BEAM]                 = 0, // TODO: Assign points
+    [EFFECT_MAX_HP_50_RECOIL]                 = 0, // TODO: Assign points
```
Comment (Bassoonian) [inline src/battle_tv.c:470]: indentation

File: test/battle/move_effect/mind_blown.c [hunk 4]
```diff
@@ -105,3 +105,16 @@ DOUBLE_BATTLE_TEST("Mind Blown causes everyone to faint in a double battle")
...
+    GIVEN {
+        PLAYER(SPECIES_CLEFAIRY) { Ability(ABILITY_MAGIC_GUARD); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_MIND_BLOWN); }
+    } SCENE {
+        NOT HP_BAR(player);
+        ANIMATION(ANIM_TYPE_MOVE, MOVE_MIND_BLOWN, player);
+    }
+}
```
Comment (Bassoonian) [inline test/battle/move_effect/mind_blown.c:120]: newline

**Comment by AlexOn1ine** [comment]
Fixed!

Today has been a day of stupid mistakes. At least some bugs are fixed 😅

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**